### PR TITLE
Style: refactor getX and generic functions

### PR DESCRIPTION
**Pull request by julianrex**
Continuation of #321, #320, #317

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] ~Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.~ Manual edit.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

- Renamed the following APIs:
    - `getLayer<T>(with:type:)` to `layer<T>(withId:type:) throws`
    - `updateLayer<T>(id:type:update:)` to `updateLayer<T>(withId:type:update:) throws`
    - `getSource<T>(id:type:)` to `source<T>(withId:type:) throws`
    - `updateGeoJSON<T>(for:with:)` to `updateGeoJSONSource<T>(withId:geoJSON:) throws`

### CODE REVIEW

File: Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/FillExtrusionLayerIntegrationTests.swift [hunk 1]
```diff
@@ -56,13 +56,11 @@ class FillExtrusionLayerIntegrationTests: MapViewIntegrationTestCase {
...
-
-            switch (retrieveResult) {
-                case .success(_):
-                    successfullyRetrievedLayerExpectation.fulfill()    
-                case .failure(let error):
-                    XCTFail("Failed to retreive FillExtrusionLayer because of error: \(error)")   
+            do {
+                _ = try style.layer(withId: "test-id", type: FillExtrusionLayer.self)
+                successfullyRetrievedLayerExpectation.fulfill()
+            } catch {
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/FillExtrusionLayerIntegrationTests.swift:62]: These are generated tests.. Is there an equivalent stylegen pr?

Comment (nishant-karajgikar) [reply Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/FillExtrusionLayerIntegrationTests.swift:62]: never mind, I see it.

**Review by neelmistry94 — APPROVED** [review]

File: Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/FillExtrusionLayerIntegrationTests.swift (hunk 1 above)
Comment (julianrex) [reply Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/FillExtrusionLayerIntegrationTests.swift:62]: There is.

File: Sources/MapboxMaps/Style/Style.swift [hunk 2]
```diff
@@ -270,12 +231,40 @@ public class Style {
...
+        let expected = closure()
+
+        if expected.isError() {
+            // swiftlint:disable force_cast
+            throw StyleError(message: expected.error as! String)
+            // swiftlint:enable force_cast
+        }
+    }
+
+    private func handleExpected<T>(closure: () -> (MBXExpected<AnyObject, AnyObject>), returnType: T.Type = T.self) throws -> T {
```
Comment (nishant-karajgikar) [inline Sources/MapboxMaps/Style/Style.swift:247]: Interesting...

**Review by nishant-karajgikar — APPROVED** [review]
LGTM!

**Pull request merged**